server/app/views/applicant/ApplicantBaseFragment.html

```diff
@@ -17,11 +17,77 @@
   <!-- This uses utext because the parameter passed into the message is html,
     it is a placeholder span with the red asterisk. We need it to be html and not
     raw text to recreate the red style -->
-  <p
-    th:utext="#{content.requiredFieldsNote(~{applicant/ApplicantBaseFragment :: requiredAsterisk})}"
-  ></p>
+  <p th:utext="#{content.requiredFieldsNote(~{this :: requiredAsterisk})}"></p>
 </div>
 
 <th:block th:fragment="requiredAsterisk">
   <span class="usa-hint--required"> *</span>
 </th:block>
+
+<div
+  th:fragment="progressBar"
+  class="usa-step-indicator usa-step-indicator--center"
+  aria-label="progress"
+>
+  <!-- The progress bar itself -->
+  <ol class="usa-step-indicator__segments">
+    <th:block
+      th:each="i: ${#numbers.sequence(0, applicationParams.totalBlockCount() - 1)}"
+      th:with="block=${applicationParams.blockList().get(i)}"
+    >
+      <!-- Only one of these two gets rendered. The first is shown when it is the
+        current step, and the other is shown for all other steps. This is made
+        as two different calls because Thymeleaf does not handle concatenation
+        with two different conditions very well, and we have 2 conditional CSS
+        classes. -->
+      <li
+        th:replace="~{this :: segmentWithLabel(${i == applicationParams.blockIndex()},
+                                               ${block},
+                                               ${i},
+                                               usa-step-indicator__segment--current)}"
+      ></li>
+      <li
+        th:replace="~{this :: segmentWithLabel(${i != applicationParams.blockIndex()},
+                                               ${block},
+                                               ${i},
+                                               ${block.isCompletedInProgramWithoutErrors()} ? 'usa-step-indicator__segment--complete' : '')}"
```

Comment:
Took me a minute but does this work?
```
      <li
        th:class="'usa-step-indicator\_\_segment' + ${i == applicationParams.blockIndex() ? ' usa-step-indicator\_\_segment--current' : ''} + ${block.isCompletedInProgramWithoutErrors() ? ' usa-step-indicator\_\_segment--complete' : ''}"
      >
        <span
          class="usa-step-indicator__segment-label"
          th:text="#{label.blockIndexLabel(${i + 1}, ${block.getName()})}"
        ></span>
      </li>
```

full file at: https://gist.github.com/avaleske/651a406263df2f60d3f145823fca817a

Reply:
That worked! I wonder why yours works when mine didn't. It might be that I wasn't escaping the underscores